Providers/Scripts/2.4x-2.5x/Scripts/nxOMSGenerateInventoryMof.py

```diff
@@ -0,0 +1,167 @@
+                        new_properties.append(property)
+
+            if len(new_properties) > 0:
+                instance['Properties'] = new_properties
+                new_instances.append(instance)
+
+    Instances = new_instances
+    
+def Set_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    init_vars(Instances)
+    Set(FileName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    return [0]
+
+def Test_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    init_vars(Instances)
+    return Test(Enable, Instances)
+
+def Get_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    arg_names = list(locals().keys())
+    init_vars(Instances)
+    
+    CurrentInstances = Instances
+    FileName = protocol.MI_String(FileName)
+    Enable = protocol.MI_Boolean(Enable)
+    for instance in CurrentInstances:
+        instance['InstanceName'] = protocol.MI_String(instance['InstanceName'])
+        instance['ClassName'] = protocol.MI_String(instance['ClassName'])
+        if instance['Properties'] is not None and len(instance['Properties']):
+            instance['Properties'] = protocol.MI_StringA(instance['Properties'])
+    Instances = protocol.MI_InstanceA(CurrentInstances)
+    RunIntervalInSeconds = protocol.MI_Uint64(RunIntervalInSeconds)
+    Tag = protocol.MI_String(Tag)
+    Format = protocol.MI_String(Format)
+    FilterType = protocol.MI_String(FilterType)
+
+    if Configuration is None:
+        Configuration = []
+    if Configuration is not None and len(Configuration):
+        Configuration = protocol.MI_StringA(Configuration)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return 0, retd
+            
+def Set(FileName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
+    if Enable == True:
+         GenerateInventoyMOF(FileName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    return [0]
+
+def Test(Enable, Instances):
+    return [-1]
+
+def GenerateInventoyMOF(FileName, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration):
+    header = '/*This is an autogenerated file \n@TargetNode=\'Localhost\'*/\n'
+    footer= '\ninstance of OMI_ConfigurationDocument \n{\n \
+    DocumentType = \"inventory\"; \n \
+    Version=\"2.0.0\"; \n \
+    MinimumCompatibleVersion = \"2.0.0\"; \n \
+    Name=\"InventoryConfig\"; \n};'
+
+    inventoryMofSection = ''
+    if Instances is not None:
+        for instance in Instances:
+            instanceName = instance['InstanceName']
```

Comment:
instanceName is not used. It should be useful if you want to add 2 instances of the same class in one file.